pkg/bootstrap/docker/dockerhelper/helper.go

```diff
@@ -72,13 +58,15 @@ func hasCIDR(cidr string, listOfCIDRs []string) bool {
 // the appropriate insecure registry argument
 func (h *Helper) HasInsecureRegistryArg() (bool, error) {
 	glog.V(5).Infof("Retrieving Docker daemon info")
-	env, err := h.client.Info()
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	info, err := h.engineAPIClient.Info(ctx)
```

Comment:
when constructed with a nil `engineAPIClient` (like `helper := dockerhelper.NewHelper(client, nil)`), this will panic